jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/MemoryMXBeanImpl.java

```diff
@@ -106,10 +106,12 @@ private static final class DisabledSharedClassProvider implements SharedClassPro
 			super();
 		}
 
+		/*[IF JAVA_SPEC_VERSION < 24]*/
 		@Override
 		public BasicPermission createPermission(String classLoaderClassName, String actions) {
```

Comment:
The import of `BasicPermission` should be conditional (here and elsewhere in this change).

Reply:
Not all of the unwanted imports have been made conditional; e.g. `SharedClassProvider.java`.

Reply:
Thanks, I updated SharedClassProvider. I took another look through and I think thats all of them.